src/python/pants/core/goals/export.py

```diff
@@ -203,11 +260,77 @@ async def export(
         raise UnrecognizedResolveNamesError(
             unexported_resolves,
             all_valid_resolve_names,
+            all_known_bin_names,
             description_of_origin="the option --export-resolve",
```

Comment:
Should this refer to the new bin options too somehow?

```suggestion
            description_of_origin="the options --export-resolve and/or --export-bin",
```

Reply:
We could even tell people which option it came from, but I think it would add more complexity than it's worth.